src/product/product.service.ts

```diff
@@ -0,0 +1,96 @@
+import {
+  BadRequestException,
+  Injectable,
+  InternalServerErrorException,
+  NotFoundException,
+} from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { validate as isUUID } from 'uuid';
+
+import { CreateProductDto, UpdateProductDto } from '@product/dto';
+import { Product } from '@product/entities/product.entity';
+import { ProductRepository } from '@product/repositories/product.repository';
+import { PaginationDto } from '@common/dto/pagination.dto';
+import { ErrorCodes } from '@common/types/error';
+
+@Injectable()
+export class ProductService {
+  constructor(
+    @InjectRepository(Product)
```

Comment:
Don't forget to delete this when you have a chance

Reply:
sure